src/hal/i2c_controller.cpp

```diff
@@ -0,0 +1,33 @@
+#include "hal/i2c_controller.hpp"
+#include "hal/i2c_wrapper.hpp"
+#include "nrf_drv_twi.h"
+#include "microbit_v2.h"
+#include "nrf_twi.h"
+#include "nrf_twi_mngr.h"
+
+namespace edge::aidan
+{
+NRF_TWI_MNGR_DEF(twi_mngr, 1, 0);
```

Comment:
not sure where exactly is the best place to put this

Reply:
looks fine. pls name constants in the future

Reply:
ill constexpr em

Reply:
macro doesnt like an int for id, not really torn up about fighting it